src/library/scala/collection/mutable/TreeSet.scala

```diff
@@ -177,7 +177,7 @@ sealed class TreeSet[A] private (private val tree: RB.Tree[A, Null])(implicit va
     // https://github.com/scala/scala/pull/4608#discussion_r34307985 for a discussion about this.
     override def foreach[U](f: A => U): Unit = iterator.foreach(f)
 
-    override def clone() = super.clone().rangeImpl(from, until)
+    override def clone(): mutable.TreeSet[A] = super.clone().rangeImpl(from, until)
```

Comment:
Can we use `self.type` in here?

Reply:
No; by definition this returns a different object, right?